src/Core/Utilities/AssemblyHelpers.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Reflection;
+
+namespace Bit.Core.Utilities
+{
+    public static class AssemblyHelpers
+    {
+        private static readonly IEnumerable<AssemblyMetadataAttribute> _assemblyMetadataAttributes;
+        private static readonly AssemblyInformationalVersionAttribute _assemblyInformationalVersionAttributes;
+        private const string GIT_HASH_ASSEMBLY_KEY = "GitHash";
+        private static string _version;
+        private static string _gitHash;
+
+        static AssemblyHelpers()
+        {
+            _assemblyMetadataAttributes = Assembly.GetEntryAssembly().GetCustomAttributes<AssemblyMetadataAttribute>();
+            _assemblyInformationalVersionAttributes = Assembly.GetEntryAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>();
+        }
+
+        public static string GetVersion()
+        {
+            if (string.IsNullOrWhiteSpace(_version))
+            {
+                _version = _assemblyInformationalVersionAttributes.InformationalVersion;
+            }
+
+            return _version;
+        }
+
+        public static string GetGitHash()
+        {
+            if (string.IsNullOrWhiteSpace(_gitHash))
+            {
+                try
+                {
+                    _gitHash = _assemblyMetadataAttributes.Where(i => i.Key == GIT_HASH_ASSEMBLY_KEY).First().Value;
+                }
+                catch (Exception)
+                { }
```

Comment:
Do we want to log something here? `Unable to retrieve the git hash`? Although I'm not certain we can inject a logger...

Reply:
Yes, I was thinking about logging. I don't think we can the "traditional" way, as static constructors cannot have any parameters. However, I'm sure we could do it a different way, but I'm not sure how you would want to handle it (ex- expecting `ILogger` in call to `GetGitHash` as a parameter there as method injection.

Reply:
Let's ignore it for now. If the value doens't exist we can return null which should indicate that the value couldn't be fetched.